pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -641,23 +678,31 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
 
 // getElasticsearchCertificateSecrets retrieves Elasticsearch certificate secrets needed for Elasticsearch to run or for
 // components to communicate with Elasticsearch. The order of the secrets returned are:
-// 1) The certificate secret needed for Elasticsearch (in the operator namespace). If the user didn't create this it is
+// 1) The certificate secret needed for ES Gateway (in the operator namespace). If the user didn't create this it is
 //    created.
-// 2) The certificate secret needed for Elasticsearch (in the Elasticsearch namespace).
-// 3) The certificate mounted by other clients that connect to Elasticsearch.
-func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (oprKeyCert *corev1.Secret, esKeyCert *corev1.Secret, certSecret *corev1.Secret, err error) {
+// 2) The certificate secret needed for ES Gateway (in the Elasticsearch namespace).
+// 3) The certificate mounted by other clients that connect to Elasticsearch (through ES Gateway).
```

Comment:
Shouldn't this be the same as 2? Because other clients should be using the Public cert, which is what a user would provide/create.

Reply:
I've changed this up quite a bit, let me know if the new function makes sense.